src/number_theory/ramification_inertia.lean

```diff
@@ -428,4 +438,235 @@ begin
     { exact is_fraction_ring.injective S L } },
 end
 
+end finrank_quotient_map
+
+section fact_le_comap
+
+local notation `e` := ramification_idx f p P
+
+/-- `R / p` has a canonical map to `S / (P ^ e)`, where `e` is the ramification index
+of `P` over `p`. -/
+noncomputable instance quotient.algebra_quotient_pow_ramification_idx :
+  algebra (R ⧸ p) (S ⧸ (P ^ e)) :=
+quotient.algebra_quotient_of_le_comap (ideal.map_le_iff_le_comap.mp le_pow_ramification_idx)
+
+@[simp] lemma quotient.algebra_map_quotient_pow_ramification_idx (x : R) :
+  algebra_map (R ⧸ p) (S ⧸ P ^ e) (ideal.quotient.mk p x) = ideal.quotient.mk _ (f x) :=
+rfl
+
+variables [hfp : fact (ramification_idx f p P ≠ 0)]
+include hfp
+
+/-- If `P` lies over `p`, then `R / p` has a canonical map to `S / P`.
+
+This can't be an instance since the map `f : R → S` is generally not inferrable.
```

Comment:
I am sure you thought about this, but why working with `f` instead of `algebra R S`?

Reply:
Using `algebra_map R S` would indeed work well enough in this part of the file, but we'd still have the `hfp : fact (ramification_idx (algebra_map R S) p P ≠ 0)` parameter meaning that it wouldn't work as an instance in general. Since `f` can be inferred (using that `fact` instace), specializing to `algebra_map` would not bring much advantage.